examples/horizontal_convection.jl

```diff
@@ -0,0 +1,347 @@
+
+# ## Turbulence closures
+#
+# We use isotropic viscosity and diffusivities, `ν` and `κ` whose values are obtain from the
+# prescribed ``Ra`` and ``Pr`` numbers. Here, we use ``Pr = 1`` and ``Ra = 10^8``:
+
+Pr = 1.0   # The Prandtl number
+Ra = 1e8   # The Rayleigh number
+
+ν = sqrt(Pr * b★ * Lx^3 / Ra)  # Laplacian viscosity
+κ = ν * Pr                     # Laplacian diffusivity
+nothing # hide
+
+# ## Model instantiation
+#
+# We instantiate the model with the fifth-order WENO advection scheme, a 3rd order
+# Runge-Kutta time-stepping scheme, and a `BuoyancyTracer`.
+
+model = IncompressibleModel(
+           architecture = CPU(),
+                   grid = grid,
+              advection = WENO5(),
+            timestepper = :RungeKutta3,
+                tracers = :b,
+               buoyancy = BuoyancyTracer(),
+                closure = IsotropicDiffusivity(ν=ν, κ=κ),
+    boundary_conditions = (b=b_bcs,)
+    )
+
+# ## Simulation set-up
+#
+# We set up a simulation that runs up to ``t = 40`` with a `JLD2OutputWriter` that saves the
+# flow speed, ``\sqrt{u^2 + w^2}``, the vorticity, ``\partial_z u - \partial_x w``, and the 
+# buoyancy dissipation, ``\kappa |\boldsymbol{\nabla} b|^2``.
+#
+# ### The `TimeStepWizard`
+#
+# The TimeStepWizard manages the time-step adaptively, keeping the Courant-Freidrichs-Lewy 
+# (CFL) number close to `0.75` while ensuring the time-step does not increase beyond the 
+# maximum allowable value for numerical stability.
+
+max_Δt = 4e-2
+wizard = TimeStepWizard(cfl=0.75, Δt=5e-3, max_change=1.2, max_Δt=max_Δt)
+
+# ### A progress messenger
+#
+# We write a function that prints out a helpful progress message while the simulation runs.
+
+CFL = AdvectiveCFL(wizard)
+
+start_time = time_ns()
+
+progress(sim) = @printf("i: % 6d, sim time: % 1.3f, wall time: % 10s, Δt: % 1.4f, CFL: %.2e\n",
+                        sim.model.clock.iteration,
+                        sim.model.clock.time,
+                        prettytime(1e-9 * (time_ns() - start_time)),
+                        sim.Δt.Δt,
+                        CFL(sim.model))
+nothing # hide
+
+# ### Build the simulation
+#
+# We're ready to build and run the simulation. We ask for a progress message and time-step update
+# every 100 iterations,
+
+simulation = Simulation(model, Δt = wizard, iteration_interval = 50,
+                                                     stop_time = 40.05,
+                                                      progress = progress)
+
+# ### Output
+#
+# We use `ComputedField`s to diagnose and output the total flow speed, the vorticity, ``\zeta``,
+# and the buoyancy dissipation, ``\chi = \kappa |\boldsymbol{\nabla}b|^2``. Note that 
+# `ComputedField`s take "AbstractOperations"on `Field`s as input:
+
+u, v, w = model.velocities # unpack velocity `Field`s
+b = model.tracers.b        # unpack buoyancy `Field`
+
+## total flow speed
+speed = ComputedField(sqrt(u^2 + w^2))
```

Comment:
4 fields might be a lot and some fields are certainly more exciting than others.  Is there a particular reason why you want all 4?  Maybe comparing wtih results from a paper?

Reply:
I wanna do some post process analysis. In the end i may chop it down to 2-3 fields